doc/compiler/ProblemDetermination.md

```diff
@@ -216,13 +216,45 @@ compilation altogether, using the `exclude=method` parameter:
 ### Identifying the failing Optimization.
 
 Most transformations that can be elided are guarded in the source code by a special
-check called `performTransformation` that can be controlled by the option
-`lastOptIndex=N`. This tells the compiler to perform `N` optional transformations,
-then proceed to only do required operations. In this sense, `lastOptIndex=-1` is an
-even stronger property than `optLevel=noOpt`.
+check called [performTransformation](https://github.com/eclipse/omr/blob/f5a790b68037be3a48b6e5c4f7e8ae6ad3e30f00/compiler/compile/OMRCompilation.hpp#L271-L274)
+([more details](https://github.com/eclipse/omr/blob/8f5212d52b533cdf9d6de9c2bd8a3e01e83e8b9d/compiler/compile/OMRCompilation.hpp#L170-L226)).
 
-By binary searching `lastOptIndex=` you can find the optimization that failed.
+`performTransformation` is [enabled by option](https://github.com/eclipse/omr/blob/8f5212d52b533cdf9d6de9c2bd8a3e01e83e8b9d/compiler/compile/OMRCompilation.hpp#L242)
+`TR_TraceOptDetails` or `TR_CountOptTransformations`.
 
+To isolate a buggy optimization, transformations can be selectively disabled by
+having `performTransformation` return false. It is (mostly) restricted to the traced
+compilations to avoid overhead in the overwhelmingly common non-tracing case.
+
+`performTransformation` is [controlled by comparing](https://github.com/eclipse/omr/blob/f5a790b68037be3a48b6e5c4f7e8ae6ad3e30f00/compiler/ras/Debug.cpp#L477)
+the current `optIndex` with `firstOptIndex` and  `lastOptIndex` options.
+
+By adjusting `N` in `lastOptIndex=N` that tells the compiler to perform `N` optional
+transformations and observing when the test passes and fails, it is possible to narrow
+down the failing transformation using a binary-search approach. Once the issue is
+narrowed down to one opt pass, then try searching for the bad transformation with
+`lastOptSubIndex`.
```

Comment:
You might also want to mention that `lastOptSubIndex` also controls `performTransformation` at this location https://github.com/eclipse/omr/blob/f5a790b68037be3a48b6e5c4f7e8ae6ad3e30f00/compiler/ras/Debug.cpp#L523-L525